### ISSUE TITLE
difficult to use with IME

### ISSUE THREAD

**Issue by o8oTo8o** [kewisch/quickmove-extension#98]
I always find it useful. Thank you very much.

By the way,
I use the IME (Input Method Editor) on Windows.
When I type into the Search box, I press Enter to determine the conversion in the IME.
This may be the reason why the first folder item is selected, even though I do not press Enter twice.
In other words, I cannot select the folder I'd like.
In previous versions, I was able to select the folder I'd like after IME conversion.
Is it possible to fix this behavior?

**Comment by kewisch** [comment]
I'll have to test and debug this, thought if you or someone else reading has some time/experience to find out what the issues is I would appreciate. I would have hoped this was pretty transparent, e.g. the first Enter does not pass through to my event listeners.

**Comment by o8oTo8o** [comment]
Thank you very much for your reply!
Unfortunately, I cannot help you.
I am not sure if this is related to the fact that there are two line feed codes (CR,LF) in Windows...

Let me change the subject a little.
In previous versions, pressing Enter on the Search box did not seem to do anything.
We had to press the down arrow to make a selection, and then press Enter to make a decision.

Perhaps with this version, the first line of the list is now selected and determined when Enter is pressed.
I think this is also an improvement.

On the other hand, the feeling of operation is very different when the target folder is the first line of the list and when it is not.
Wouldn't it be more likely to select the first line by mistake?

In other words, I would appreciate a debug on the IME, though, I think there is a way to revert back to the specification where you cannot make a decision before moving to the list.

It would be helpful if you could debug this in one of these ways.
Thanks a lot!

**Comment by kewisch** [comment]
Can you try this version?

[quick_folder_move-2.6.6.zip](https://github.com/kewisch/quickmove-extension/files/12835775/quick_folder_move-2.6.6.zip)

**Comment by o8oTo8o** [comment]
I tried ver.2.6.6.
However, the behavior did not change.

**Comment by kewisch** [comment]
Alright. I'm guessing it doesn't get better with 2.6.9 also posted there. Can you describe how I can trigger this bug to a person who has never knowingly used IME before nor speaks the languages commonly associated with it? (e.g. what keyboard setting to use, what keys to press, etc)

**Comment by o8oTo8o** [comment]
It certainly doesn't seem to be a problem with 1 byte character IMEs.
I triggered this bug when I typed Japanese.

On Windows, you can install an Japanese "Microsoft IME" as follows.
Settings > Time & Language > Language & region
<1>   Language > Add a language > Japanese
<2>   Language > Japanese > Language options > Keyboards > Add a keyboard > Microsoft IME

Switch IME to "Microsoft IME(Japanese)".
When you switch the IME to "Microsoft IME(Japanese)", "A" and "J in circle" will appear on the taskbar.
To activate or deactivate the input of Japanese characters, press the '`' key.
When activated, the letter "A" on the tasker becomes "あ".
If you type "a", "あ over wavy line" will be displayed, and "あ" will be determined by continuing to enter.

Captured images are attached.
IME01.png: Before typing into the Search box
IME02.png: In the process of converting with IME ("あ over wavy line")
IME03.png: After typing Enter for decision
![IME01](https://github.com/kewisch/quickmove-extension/assets/144748707/f6a81b0a-e5d4-4c37-b316-3764f3e86b74)
![IME02](https://github.com/kewisch/quickmove-extension/assets/144748707/2d7208c1-12df-4235-bb00-3e5efabef02d)
![IME03](https://github.com/kewisch/quickmove-extension/assets/144748707/d24eb70e-a6bc-4cce-aa4e-ab9a06b8b8dc)

**Comment by kewisch** [comment]
@o8oTo8o how is this? With a little more reading and my windows machine fired up I finally got around to this. I believe it should work. I was able to avoid the enter command firing. The test folders I created with Japanese symbols, I was not able to find them by entering the characters, even with copy paste. Either there are further bugs, or I am doing it wrong.

Could you give this version a test run?


[quick_folder_move-2.9.3pre.zip](https://github.com/user-attachments/files/17877459/quick_folder_move-2.9.3pre.zip)

**Comment by o8oTo8o** [comment]
Great!
It works well.
Thanks a lot.

**Comment by kewisch** [comment]
Sweet, thank you!  I'll include this in the next version

c13b619d83d145e6a26e44c5ef3b2dd15c300068

**Comment by kewisch** [comment]
@o8oTo8o could you give this one a try? I needed to make some changes due to another feature which basically removes the change I made previously, but it should work because it reacts to a different key event.

[quick_folder_move-3.0.0.zip](https://github.com/user-attachments/files/18060272/quick_folder_move-3.0.0.zip)
